packages/cli/src/commands/site/deploy/siteDeployCommand.mts

```diff
@@ -88,41 +76,36 @@ export default async function siteDeployCommand(
 
   consola.start("Uploading site files");
   await Promise.all([
-    artifacts.SiteAssetArtifactsService.uploadZippedSiteAsset(
+    thirdPartyApplications.uploadWebsiteVersion(
       clientCtx,
-      {
-        application,
-        version: siteVersion,
-        zipFile: Readable.toWeb(archive) as ReadableStream<any>, // This cast is because the dom fetch doesnt align type wise with streams
-      },
+      application,
+      siteVersion,
+      Readable.toWeb(archive) as ReadableStream<any>, // This cast is because the dom fetch doesnt align type wise with streams
     ),
     archive.finalize(),
   ]);
   consola.success("Upload complete");
 
-  const repositoryRid = await thirdPartyApplicationService
-    .fetchWebsiteRepositoryRid(clientCtx, application);
-  const registeredSiteDomains = await ArtifactsSitesAdminV2Service
-    .getRegisteredSiteDomains(
-      ctx,
-      repositoryRid,
-    );
-  const domain = getFirstSiteDomain(registeredSiteDomains);
-
   if (!uploadOnly) {
-    await ArtifactsSitesAdminV2Service.updateDeployedVersion(
-      ctx,
-      repositoryRid,
-      { siteVersion: { version: siteVersion } },
+    const deployment = await thirdPartyApplications.updateWebsiteDeployment(
+      clientCtx,
+      application,
+      { version: siteVersion },
     );
     consola.success(`Deployed ${siteVersion} successfully`);
+    const domain = deployment.subdomains[0];
     if (domain != null) {
       logSiteLink(
         "View live site:",
         `https://${domain}`,
       );
     }
   } else {
+    const deployment = await thirdPartyApplications.getWebsiteDeployment(
```

Comment:
There's one difference here where the deployment contains the domains meaning if no version is currently deployed we can no longer show the preview URL with the uploaded version